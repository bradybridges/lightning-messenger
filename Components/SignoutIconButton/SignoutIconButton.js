import React from 'react';
import {
  TouchableOpacity,
  Image,
  StyleSheet,
  Dimensions,
} from 'react-native';
import * as firebase from 'firebase';
import 'firebase/auth';

const logoutIcon = require('../../assets/logout.png');

const SignoutIconButton = () => (
  <TouchableOpacity style={styles.container} onPress={() => firebase.auth().signOut()}>
    <Image style={styles.img} source={logoutIcon} />
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    height: Dimensions.get('window').height * 0.08,
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: 10,
  },
  img: {
    width: 30,
    height: 30,
  },
});

export default SignoutIconButton;
