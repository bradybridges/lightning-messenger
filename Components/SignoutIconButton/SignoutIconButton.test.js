import React from 'react';
import { shallow } from 'enzyme';
import SignoutIconButton from './SignoutIconButton';

describe('SignoutIconButton', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<SignoutIconButton />);
    expect(wrapper).toMatchSnapshot();
  });
});
