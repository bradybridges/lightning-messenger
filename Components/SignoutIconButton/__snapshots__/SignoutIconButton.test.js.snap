// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignoutIconButton should match snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": 106.72,
      "justifyContent": "center",
      "marginRight": 10,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 30,
        "width": 30,
      }
    }
  />
</TouchableOpacity>
`;
